### ISSUE TITLE
failed to initialize project with "go.kubebuilder.io/v3"

### ISSUE THREAD

**Issue by abhi1693** [kubernetes-sigs/kubebuilder#1971]
```                                                                                                                
➜  kubeact-operator git:(feat/v0.0.1) ✗ kubebuilder init                                                                                        git:(feat/v0.0.1|●53✚1…2 
Error: failed to initialize project with "go.kubebuilder.io/v3": only the go.mod and files with the prefix "(.)" are allowed before the init
Usage:
  kubebuilder init [flags]

Examples:
  # Scaffold a project using the apache2 license with "The Kubernetes authors" as owners
  kubebuilder init --project-version=2 --domain example.org --license apache2 --owner "The Kubernetes authors"


Flags:
      --component-config         create a versioned ComponentConfig file, may be 'true' or 'false'
      --domain string            domain for groups (default "my.domain")
      --fetch-deps               ensure dependencies are downloaded (default true)
  -h, --help                     help for init
      --license string           license to use to boilerplate, may be one of 'apache2', 'none' (default "apache2")
      --owner string             owner to add to the copyright
      --plugins strings          Name and optionally version of the plugin to initialize the project with. Available plugins: ("go.kubebuilder.io/v2", "go.kubebuilder.io/v3")
      --project-name string      name of this project
      --project-version string   project version, possible values: ("2", "3-alpha") (default "3-alpha")
      --repo string              name to use for go module (e.g., github.com/user/repo), defaults to the go package of the current working directory.
      --skip-go-version-check    if specified, skip checking the Go version

2021/01/24 07:14:18 failed to initialize project with "go.kubebuilder.io/v3": only the go.mod and files with the prefix "(.)" are allowed before the init
➜  kubeact-operator git:(feat/v0.0.1) ✗ ls                                                                                                                                       git:(feat/v0.0.1|●53✚1…2 
go.mod  go.sum
➜  kubeact-operator git:(feat/v0.0.1) ✗ 
```

* go version go1.15.6 linux/amd64
* Version: main.version{KubeBuilderVersion:"3.0.0-alpha.1", KubernetesVendor:"1.19.2", GitCommit:"1428bba7c162cf2b783c89fd6f07f4efbae97130", BuildDate:"2021-01-08T22:30:36Z", GoOs:"linux", GoArch:"amd64"}

/kind bug

**Comment by daenney** [comment]
You have a `go.sum` in that directory it seems, which would explain the error since it seems to only permit `go.mod`?

But I get the exact same error without a `go.sum` present, so this seems broken one way or another.

**Comment by daenney** [comment]
The error also triggers in a new git repo without anything present but a `.git` when the plugin is invoked with `--repo`.

**Comment by camilamacedo86** [comment]
See the error `failed to initialize project with "go.kubebuilder.io/v3": only the go.mod and files with the prefix "(.)" are allowed before the init`. It means that your directory has files already which are not go.mod or that ones which starts with (.)

**Comment by abhi1693** [comment]
@camilamacedo86  So, I cannot initialize a project which is checked out of Github? This does not make sense as I have a template with which I normally start a new project and I cannot use it due to this issue. In my knowledge, `kubebuilder` does not use any `dotfile` so why put such a limitation on the initialization?

I have several dotfiles and folders that are part of our automation. And the error states that dotfiles and dot-folders are indeed allowed. So, this is a bug with either the code or the message.

**Comment by camilamacedo86** [comment]
Kubebuilder should allow initing the project in a directory with files/directories which starts with (.) and that has go.mod already. However, other directories not because then we cannot ensure that the tool will do the scaffold properly. You can init the project and then,  push your changes to the top as all. 

Are you facing issues when you have a file/directory whcih has the prefix (.)? If yes, we need to fix that. Could you check my comments in your PR and change the fix to ensure that only these scenarios will be allowed? Also, please let me know if it makes sense for you or not. 

Also, please feel free to raise your suggestions regards to it.

**Comment by abhi1693** [comment]
@camilamacedo86 I tried again with only .git and go.mod and it still failed with the same issue

**Comment by Adirio** [comment]
This bug was already discovered and #1944 is working on fixing it. Sorry for the inconvenience.

**Comment by estroz** [comment]
IIRC it was decided that initializing a new project with `go.mod` is _not_ supported because `make` could fail.

**Comment by Adirio** [comment]
This should be fixed by #1944, if you find that it is still bugged please reopen this issue.

/close
